pkg/activator/handler/requestevent_handler_test.go

```diff
@@ -69,3 +69,35 @@ func TestRequestEventHandler(t *testing.T) {
 		t.Errorf("out.Time.Sub(in.Time) = %v, want > %v", out.Time.Sub(in.Time), timeOffset)
 	}
 }
+
+func BenchmarkRequestEventHandler(b *testing.B) {
+	baseHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
+	reqCtx := util.WithRevID(context.Background(), types.NamespacedName{Namespace: testNamespace, Name: testRevName})
+
+	reqChan := make(chan network.ReqEvent, 100)
+	defer close(reqChan)
+	go func() {
+		// Just consume all the events.
+		for range reqChan {
+		}
+	}()
+
+	handler := NewRequestEventHandler(reqChan, baseHandler)
+
+	resp := httptest.NewRecorder()
+	b.Run("sequential", func(b *testing.B) {
+		req := httptest.NewRequest(http.MethodGet, "http://example.com", nil).WithContext(reqCtx)
```

Comment:
just out of interest, is there any reason not to share req between the sequential and parallel runs like we do with resp?

Reply:
IIRC I did that in other benchs to move the ref into the `RunParallel` call, so there's no race between parallel accesses. IIRC those benchs mutated the req though, I can check if that's the case here too.